smac/multi_objective/utils.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import annotations
+
+
+def normalize_costs(values: list[float], bounds: list[tuple[float, float]] | None = None) -> list[float]:
+    """
+    Normalizes a list of floats with corresponding bounds.
+
+    Parameters
+    ----------
+    values : list[float]
+        List of costs to be normalized.
+    bounds : list[tuple[float, float]] | None, optional
+        List of tuple of bounds. By default None. If no bounds are passed, the values are returned
+        unnormalized.
+
+    Returns
+    -------
+    normalized_costs : list[float]
+        Normalized costs based on the bounds. If no bounds are given, the values are returned unnormalized.
+        Also, if min and max bounds are the same, the value of the corresponding objective is set to 1.
+    """
+    if bounds is None:
+        return values
+
+    if len(values) != len(bounds):
+        raise ValueError("Number of values and bounds must be equal.")
+
+    costs = []
+    for v, b in zip(values, bounds):
```

Comment:
Why not transform them into `np.array` and transform the cost array back into `List[float]` (if it needs to be stored as `List`)

Reply:
Is there any benefit doing it so?

Reply:
Hmmm... then at least you could preserve the space for `cost`, something like `costs = [0.] * len(values)`

Reply:
why?